### PR TITLE
GCW-3427 Adds Process actions can link to a source

### PR DESCRIPTION
**Pull request by bharat619**
### Ticket Link: 
https://jira.crossroads.org.hk/browse/GCW-3427

### What does this PR do?
Add source to Process actions

### CODE REVIEW

**Review by shreyas098 — APPROVED** [review]

File: app/models/processing_destinations_lookup.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+export default Model.extend({
+  name: attr("string"),
+  lookups: function() {
+    const allRecords = this.store.peekAll("processing_destinations_lookup");
+    return allRecords.map(record => ({
+      id: record.get("id"),
+      name: record.get("name")
+    }));
+  }
```
Comment (patrixr) [inline app/models/processing_destinations_lookup.js:12]: Don't think that `lookups` method has its place in the model (also did a search, doesn't seem to be used?)

File: app/routes/items/detail.js [hunk 2]
```diff
@@ -40,6 +40,9 @@ export default AuthorizeRoute.extend({
     }
     await this.store.findAll("restriction", { reload: true });
     await this.store.findAll("donor_condition", { reload: true });
+    await this.store.findAll("processing_destinations_lookup", {
+      reload: true
+    });
```
Comment (patrixr) [inline app/routes/items/detail.js:45]: All 3 of those here probably don't need to be reloaded every time we open item details.
Either:
- We preload them
- We load them when we open the items details page, but only the first time (cache them essentially)

File: app/mixins/item_actions.js [hunk 3]
```diff
@@ -186,6 +197,10 @@ export default Ember.Mixin.create(AsyncMixin, {
       }
     },
 
+    onProcessingDestinationChange(destination) {
```
Comment (patrixr) [inline app/mixins/item_actions.js:200]: `setProcessingDestination` makes more sense as a name

`onXXX` would mean that it has already happened, but this method actually does the set

**Review by patrixr — CHANGES_REQUESTED** [review]
Looks good, few minor comments

File: app/models/processing_destinations_lookup.js (hunk 1 above)
Comment (bharat619) [reply app/models/processing_destinations_lookup.js]: yeah. I had added it as some experimentation. Then forgot to remove. Thanks, removed it

File: app/routes/items/detail.js (hunk 2 above)
Comment (bharat619) [reply app/routes/items/detail.js]: cached it

**Review by patrixr — APPROVED** [review]

**Pull request merged**